### ISSUE TITLE
[P0] Duplicate case labels not detected in switch

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#20]
## Problem

The compiler silently accepts duplicate `case` labels in a `switch` statement. C11 §6.8.4.2p3 requires that each case constant expression in a switch be unique after conversion. GCC and Clang both emit `error: duplicate case value`.

## Expected behavior

```
test.c:4:5: error: duplicate case value '1'
test.c:3:5: note: previous case defined here
```

## Reproduction

```c
int main(void) {
    int x = 1;
    switch (x) {
        case 1: return 1;
        case 1: return 2;  // should error: duplicate case value
    }
    return 0;
}
```

Compile with `ccc -S -o /dev/null test.c` — exits 0, no diagnostic.

## Suggested approach

The best place to detect this is in **sema** (`src/frontend/sema/analysis.rs`), not in the lowering phase. Add a `switch_case_values` stack (similar to how the lowerer uses `switch_stack`) that tracks seen case values per switch level.

In `analyze_stmt` for `Stmt::Switch`:
1. Push a new `HashSet<i64>` (or `HashMap<i64, Span>`) onto a `switch_cases` stack before analyzing the body
2. Pop it after

In `analyze_stmt` for `Stmt::Case`:
1. Evaluate the case constant using sema's const eval (`SemaConstEval`)
2. Check if it's already in the current switch's set
3. If duplicate, emit `error: duplicate case value '{}'` with a note pointing to the first occurrence
4. Otherwise, insert it

## Files to modify

- `src/frontend/sema/analysis.rs` — Add `switch_cases: Vec<HashMap<i64, Span>>` to `SemanticAnalyzer`, push/pop in `Stmt::Switch`, check/insert in `Stmt::Case`

## Tests

Add tests in a `#[cfg(test)] mod tests` block at the bottom of `analysis.rs`. You'll need a test helper that compiles C code through sema and returns the error count. Pattern:

```rust
fn sema_error_count(code: &str) -> usize {
    // Parse code, run SemanticAnalyzer::analyze(), return error count
}

#[test]
fn duplicate_case_label() {
    let errors = sema_error_count("int f(void) { switch(0) { case 1: break; case 1: break; } return 0; }");
    assert!(errors > 0);
}

#[test]
fn distinct_case_labels_ok() {
    let errors = sema_error_count("int f(void) { switch(0) { case 1: break; case 2: break; } return 0; }");
    assert_eq!(errors, 0);
}
```

## Acceptance criteria

- [ ] `ccc -S -o /dev/null test.c` emits an error for duplicate case values
- [ ] Error message includes the duplicate value
- [ ] Clean build: `cargo build --release`
- [ ] All existing tests pass: `cargo test --lib`
- [ ] New tests added for the fix
